src/components/_Controllers/CMSPreview.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react';
+import { Route, Switch } from 'react-static';
+import PropTypes from 'prop-types';
+import { injectIntl } from 'react-intl';
+import { request } from 'graphql-request';
+import { createGraphQLClientsByLang } from 'js/helpers/fetchData';
+import allServicePagesQuery from 'js/queries/allServicePagesQuery';
+import allProcessesQuery from 'js/queries/allProcessesQuery';
+import { cleanProcesses, cleanServices } from 'js/helpers/cleanData';
+import Services from 'components/Pages/Services';
+import Processes from 'components/Pages/Processes';
+
+class CMSPreview extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: null,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData() {
+    const {
+      intl,
+      match: {
+        params: { revision_id, page_type },
+      },
+    } = this.props;
+
+    const client = createGraphQLClientsByLang(intl.locale);
+    //TODO: one endpoint for revisions data requests
+    let req;
+    switch (page_type) {
+      case 'services':
+        req = client.request(allServicePagesQuery);
+        break;
+      case 'processes':
+        req = client.request(allProcessesQuery);
+        break;
```

Comment:
This switch statement is making me think it would be nice to have some documentation or checklist about what we need to change in Joplin whenever we add a new content type. Just a thought so we don't forget something in the future.

Reply:
I agree -- do you think the ReadMe is a good place for this?

Reply:
Actually, the more I think about it, we shouldn't preserve this switch statement long term as it couples our front end logic to the CMS -- we should rely on one request endpoint, that delivers the data formatted for the appropriate content-type. What do you think?

Reply:
ah yes, that does seem like a better long term solution.